src/Payload.cpp

```diff
@@ -55,6 +55,18 @@ void Payload::clear() {
   metadata.reset();
 }
 
+Payload Payload::clone() const {
+	Payload out;
```

Comment:
I don't agree with clearer, you've packed two ternary operators into a variable intialisation.